examples/datasets/allstate.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2021 IBM Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import subprocess
+from scipy.sparse import save_npz, load_npz
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import MinMaxScaler, LabelEncoder, OneHotEncoder, normalize
+from scipy.sparse import save_npz, load_npz, csr_matrix
+
+class Allstate():
+
+    def __init__(self, cache_dir):
+        self.name = type(self).__name__
+        self.cache_dir = cache_dir
+        self.working_dir = os.path.join(self.cache_dir, self.name)
+        self.files = ['allstate.X_train.npy', 
+                      'allstate.y_train.npy', 
+                      'allstate.X_test.npy', 
+                      'allstate.y_test.npy']
+
+    def __check_files_exist(self, files):
+        files_exist = True
+        for file in files:
+            files_exist &= os.path.isfile(os.path.join(self.working_dir, file))
+        return files_exist
+
+    def get_train_test_split(self):
+
+        if not self.__check_files_exist(self.files):
+
+
+            p = subprocess.Popen(['mkdir', '-p', self.working_dir])
+            p.wait()
+
+            p = subprocess.Popen(['kaggle', 'competitions', 'download', '-c', 'ClaimPredictionChallenge'], cwd=self.working_dir)
+            p.wait()
+
+            p = subprocess.Popen(['unzip', 'ClaimPredictionChallenge.zip'], cwd=self.working_dir)
+            p.wait()
+            
+            p = subprocess.Popen(['unzip', 'train_set.zip'], cwd=self.working_dir)
+            p.wait()
+            
+
+            df = pd.read_csv(os.path.join(self.working_dir, 'train_set.csv'))
+            df.drop(['Row_ID'],axis=1,inplace=True)
+            df.replace('?', np.NaN, inplace=True)
+            df.fillna(-1, inplace=True)
+
+            df_Y = df[['Claim_Amount']]
```

Comment:
we need to make sure that we don't reuse code from external sources, e.g., for preprocessing. this comments is valid for all datasets.

Reply:
All of the other datasets are our own code except this one. I asked you whether it came from somewhere else and you pointed me to the publicly available notebook written by the India team.

Reply:
Then I guess we're safe. That code is under IBM license so it's fine, we can reuse it.